src/mock_etcd.rs

```diff
@@ -76,13 +83,31 @@ impl MockEtcdServer {
     }
 }
 
+/// `KeyRange` is an abstraction for describing etcd key of various types.
+struct KeyRange {
+    /// The first key of the range and should be non-empty
+    key: Vec<u8>,
+    /// The key following the last key of the range
+    range_end: Vec<u8>,
+}
+
 /// Mock Etcd
-#[derive(Debug, Clone)]
+#[derive(Clone)]
 struct MockEtcd {
     /// map to store key value
     map: Arc<RwLock<HashMap<Vec<u8>, KeyValue>>>,
+
+    /// map to store key and watch ids
+    watch: Arc<RwLock<HashMap<i64, KeyRange>>>,
+
+    /// map to store watch id and watch response senders
+    #[allow(clippy::type_complexity)]
+    watch_response_sender: Arc<RwLock<HashMap<i64, Arc<RwLock<DuplexSink<WatchResponse>>>>>>,
```

Comment:
We'd better define the type with a name, it's hard to understand for such a complex type.
For example:
``` rust
type LockedDuplexSink = Arc<RwLock<DuplexSink<WatchRespnse>>>;
```

Reply:
If we only have write operation for the DuplexSink, use Mutex instead of a Rwlock.